### ISSUE TITLE
Zmachine can claim any public ip without reserving it

### ISSUE THREAD

**Issue by OmarElawady** [threefoldtech/zos#1457]
After reserving a machine with a public ip `185.206.122.46`. The user can add another public ip to the eth1 interface (`ip addr add ...`) and it will be announced to the network and curling it from outside arrives to the zmachine.

**Comment by muhamadazmy** [comment]
I remember we had firewall rules only to prevent this. I will look into it.

**Comment by muhamadazmy** [comment]
This issue is confirmed

We have nft rules to prevent traffic that is not originated from this IP+MAC combination assigned to the VM on creation, so changing the mac, and or the Ip will render the node un-reachable.

The problem we confirmed is that those rules are now "not working" the counters says the traffic going over the device does not hit the rule at all.

We suspect that this might relate to using macvtap instead of taps . we switched to macvtaps a while back because we had issues with cloud-hypervisor. We might need to test this again with latest release of cloud-hypervisor
